service-base/src/main/java/org/eclipse/hono/service/metric/MetricsTags.java

```diff
@@ -241,6 +243,82 @@ public Tag asTag() {
         }
     }
 
+    /**
+     * A status providing more information about the outcome of processing a message received from a device.
+     * <p>
+     * The immutable enum check is disabled because the offending field's (tag) type
+     * is in fact immutable but has no corresponding annotation.
+     */
+    @SuppressWarnings("ImmutableEnumChecker")
+    public enum Reason {
```

Comment:
I think we can be more explicit concerning the `enum` name here. ('Reason' for what?)
How about `ProcessingOutcomeReason`?